core/src/game/mainLogic.ts

```diff
@@ -0,0 +1,82 @@
+import { Board } from "../board/boardType";
+import {
+  GameState,
+  checkIfGameWon,
+  updateRemainingTiles,
+  winGame,
+} from "./state";
+import { highlightAllowedTiles, updateBoardState } from "../board";
+import { getOppositePath, updateGraphState } from "../graph";
+import { switchPlayer } from "../player";
+import { PlayableTile } from "../tile";
+import { checkUserMove } from "../move";
+import { checkIfDraw } from "./state";
+import { setGameAsDraw } from "./state";
+
+export const mainLogic = (
```

Comment:
Really not sure of the naming. What does this function is supposed to do? Do it try to place a tile? `mainLogic` is to generic. You can find a better one 😉